src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py

```diff
@@ -168,6 +169,9 @@ def collectAgentInfo(self):
             if float(disk['percent'].strip('%')) >= diskUseThreshold and \
                             disk['mounted'] not in self.config.AnalyticsDataCollector.ignoreDisk:
                 agentInfo['disk_warning'].append(disk)
+                agentInfo['drain_mode'] = True
+                self.config.WorkQueueManager.queueParams['DrainMode'] = True
+                os.system("$manage execute-agent wmcoreD --restart --components=WorkQueueManager,AnalyticsDataCollector,AgentStatusWatcher")
```

Comment:
Giving it some thought, I think it would be useful to have a ManageAgent or something like that under
https://github.com/dmwm/WMCore/tree/master/src/python/WMCore/Agent

with basically the same functionality as wmcoreD. I'm sure it would be used for other purposes once we get it.
@ticoann what do you think?